Sources/MUXSDKStatsObjc/MUXSDKMonitor.h

```diff
@@ -0,0 +1,397 @@
+/// video, and view metadata
+///   - automaticErrorTracking: boolean to indicate if the
+/// SDK should automatically track player errors
+/// - Returns: an instance of MUXSDKPlayerBinding or nil
+- (nullable MUXSDKPlayerBinding *)startMonitoringPlayer:(nonnull AVPlayer *)player
+                                         withPlayerName:(nonnull NSString *)playerName
+                                        fixedPlayerSize:(CGSize)fixedPlayerSize
+                                           customerData:(nonnull MUXSDKCustomerData *)customerData
+                                 automaticErrorTracking:(BOOL)automaticErrorTracking;
+
+/// Starts to monitor a given AVPlayer instance.
+/// The instance must be identified with a globally unique
+/// player name.
+///
+/// - Parameters:
+///   - player: An AVPlayer to monitor
+///   - playerName: A globally unique name for this instance
+///   of the player
+///   - fixedPlayerSize: A fixed size of your player that will
+///   not change, inclusive of any letter boxed or pillar
+///   boxed areas. If monitoring audio only media, pass in
+///   `CGSizeMake(0.0, 0.0)`
+///   - customerData: A MUXSDKCustomerData object with player,
+/// video, and view metadata
+///   - automaticErrorTracking: boolean to indicate if the
+/// SDK should automatically track player errors
+///   - beaconCollectionDomain: Domain to send tracking data
+///   to, if you want to use a custom beacon domain. Optional.
+/// - Returns: an instance of MUXSDKPlayerBinding or nil
+- (nullable MUXSDKPlayerBinding *)startMonitoringPlayer:(nonnull AVPlayer *)player
+                                         withPlayerName:(nonnull NSString *)playerName
+                                        fixedPlayerSize:(CGSize)fixedPlayerSize
+                                           customerData:(nonnull MUXSDKCustomerData *)customerData
+                                 automaticErrorTracking:(BOOL)automaticErrorTracking
+                                 beaconCollectionDomain:(nullable NSString *)beaconCollectionDomain;
+
+/// Updates the AVPlayer instance being monitored
+/// for the provided player name.
+///
+/// Monitoring must have already been started for the 
+/// player name.
+///
+/// - Parameters:
+///   - player: The new AVPlayer to monitor
+///   - playerName: The name of the player instance to update
+///   - fixedPlayerSize: A fixed size of your player that will
+///   not change, inclusive of any letter boxed or pillar
+///   boxed areas. If monitoring audio only media, pass in
+///   `CGSizeMake(0.0, 0.0)`
+- (void)updatePlayer:(nonnull AVPlayer *)player
+      withPlayerName:(nonnull NSString *)playerName
+     fixedPlayerSize:(CGSize)fixedPlayerSize;
+
+#pragma mark - Stop Monitoring
+
+/// Removes any observers monitoring the instance with the
+/// provided player name
+///
+/// - Parameter playerName: The player name to stop monitoring
+- (void)stopMonitoringWithPlayerName:(nonnull NSString *)playerName;
```

Comment:
Objective-C lacks type-based overloading (unlike Swift) so the `WithPlayerName:` is required here unless `stopMonitoringWithPlayerBinding:` also gets dropped.